resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientSuite.scala

```diff
@@ -35,18 +36,22 @@ import org.apache.hadoop.yarn.conf.YarnConfiguration
 import org.apache.hadoop.yarn.util.Records
 import org.mockito.Matchers.{eq => meq, _}
 import org.mockito.Mockito._
-import org.scalatest.Matchers
+import org.scalatest.{BeforeAndAfterAll, Matchers}
 
 import org.apache.spark.{SparkConf, SparkFunSuite, TestUtils}
 import org.apache.spark.deploy.yarn.config._
 import org.apache.spark.util.{SparkConfWithEnv, Utils}
 
-class ClientSuite extends SparkFunSuite with Matchers {
+class ClientSuite extends SparkFunSuite with Matchers with BeforeAndAfterAll {
 
   import Client._
 
   var oldSystemProperties: Properties = null
 
+  override def beforeAll(): Unit = {
+    super.beforeAll()
```

Comment:
doesn't seem like you're making use of `BeforeAndAfterAll`?

Reply:
I don't quite remember why I needed this, but it was related to the call to System.setProperty("spark.testing", "true") in SparkFunSuite.beforeAll.